### PR TITLE
SER-3485 | Recreate the wiki inclusion/exclusion rules in unified-search: article search

### PR DESCRIPTION
**Pull request by mmeller-wikia**
https://wikia-inc.atlassian.net/browse/SER-3485
Add data for search indexing rules

### CODE REVIEW

File: extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php [hunk 1]
```diff
@@ -49,7 +49,30 @@ public function execute() {
...
+		$result['is_disabled_in_global_search'] = $this->getWikiGlobalSearchInclusionStats();
 		return $result;
 	}
 
+	/**
+	 * Calculate flag for deciding to include page in global search.
+	 *
+	 * @return array
+	 */
+	protected function getWikiGlobalSearchInclusionStats() {
```
Comment (mateuszsip) [inline extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php]: we could rename it to `isDisabledInGlobalSearch`

File: extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php [hunk 2]
```diff
@@ -49,7 +49,30 @@ public function execute() {
...
+		global $wgExcludeWikiFromSearch;
+		global $wgForceWikiIncludeInSearch;
+
+		if (!empty( $wgExcludeWikiFromSearch )) {
```
Comment (mateuszsip) [inline extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php]: I'm not sure if formatting is not correct - [here](https://github.com/Wikia/guidelines/tree/master/PHP) in bin directory you can find xml with IDE config

**Review by mateuszsip — APPROVED** [review]
Thanks!

File: extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php [hunk 3]
```diff
@@ -49,7 +49,30 @@ public function execute() {
...
+		if ( !empty( $wgExcludeWikiFromSearch ) ) {
+			return true;
+		}
+		if ( !empty( $wgForceWikiIncludeInSearch ) ) {
+			return false;
+		}
+
+		$service = $this->getService();
+		$wikiData = $service->getApiStatsForWiki();
+		$wikiArticles = $wikiData['query']['statistics']['articles'];
```
Comment (mszabo-wikia) [inline extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php]: kinda overkill, `SiteStats::articles` does the same

Comment (mmeller-wikia) [reply extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php]: Thanks :)

**Review by pwojcik86 — APPROVED** [review]

**Pull request merged**